### PR TITLE
Translate index.md to Turkish

### PR DESCRIPTION
**Pull request by mertyyanik**
# What does this PR do?

Translated `index.md` to Turkish language.
Part of #27088 

## Who can review?
@stevhliu, @MKhalusova, @merveenoyan

### CODE REVIEW

**Comment by MKhalusova** [comment]
Hi @mertyyanik ! Thank you for working on translating the doc page into Turkish.
From what I see, this is the first time a doc page has been translated into Turkish. As such, you'll need to modify two more files in order to build the docs in a new language - .github/workflows/build_documentation.yml and .github/workflows/build_pr_documentation.yml.  

Add the two-letter code for your language to the list of languages. You can look up the code [here](https://www.loc.gov/standards/iso639-2/php/code_list.php).
Also pinging @merveenoyan for a review of the Turkish translation :)

**Comment by mertyyanik** [comment]
I made the changes. Thanks @MKhalusova and @merveenoyan

File: docs/source/tr/_toctree.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+- sections:
+  - local: index
+    title: 🤗 Transformers
```
Comment (stevhliu) [inline docs/source/tr/_toctree.yml:3]:
```suggestion
    title: 🤗 Transformers
  title: Get started
```

**Review by stevhliu — APPROVED** [review]
Thanks for starting the Turkish translations! The format and structure look good to me, I'll let @merveenoyan comment on the actual content itself 🙂

**Comment by mertyyanik** [comment]
Hello, @merveenoyan. I need a review. When it's done, I will proceed to the other sections.

File: docs/source/tr/index.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+http://www.apache.org/licenses/LICENSE-2.0
+
+Lisansa tabi olmayan durumlarda veya yazılı anlaşma olmadıkça, Lisans kapsamında dağıtılan yazılım, herhangi bir türde (açık veya zımni) garanti veya koşul olmaksızın, "OLDUĞU GİBİ" ESASINA GÖRE dağıtılır. Lisans hükümleri, özel belirli dil kullanımı, yetkileri ve kısıtlamaları belirler.
+
+⚠️ Bu dosya Markdown biçimindedir, ancak belge oluşturucumuz için özgü sözdizimleri içerir (MDX gibi) ve muhtemelen Markdown görüntüleyicinizde düzgün bir şekilde görüntülenmeyebilir.
+-->
+
+# 🤗 Transformers
+
+[PyTorch](https://pytorch.org/), [TensorFlow](https://www.tensorflow.org/) ve [JAX](https://jax.readthedocs.io/en/latest/) için State-of-the-art Makine Öğrenimi.
```
Comment (merveenoyan) [inline docs/source/tr/index.md:14]:
```suggestion
[PyTorch](https://pytorch.org/), [TensorFlow](https://www.tensorflow.org/) ve [JAX](https://jax.readthedocs.io/en/latest/) için son teknoloji makine öğrenimi.
```

File: docs/source/tr/index.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+🤗 Transformers, güncel önceden eğitilmiş (pretrained) modelleri indirmenizi ve eğitmenizi kolaylaştıran API'ler ve araçlar sunar. Önceden eğitilmiş modeller kullanarak, hesaplama maliyetlerinizi, karbon ayak izinizi azaltabilir ve sıfırdan bir modeli eğitmek için gereken zaman ve kaynaklardan tasarruf edebilirsiniz. Bu modeller farklı modalitelerde ortak görevleri destekler, örneğin:
+
+📝 **Doğal Dil İşleme**: metin sınıflandırma, adlandırılmış varlık tanıma, soru cevaplama, dil modelleme, özetleme, çeviri, çoktan seçmeli ve metin oluşturma.<br>
+🖼️ **Bilgisayar Görüsü**: görüntü sınıflandırma, nesne tespiti ve bölümleme (segmentation).<br>
```
Comment (merveenoyan) [inline docs/source/tr/index.md:19]:
```suggestion
🖼️ **Bilgisayarlı Görü**: görüntü sınıflandırma, nesne tespiti ve bölümleme (segmentation).<br>
```

File: docs/source/tr/index.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+🗣️ **Ses**: otomatik konuşma tanıma ve ses sınıflandırma.<br>
+🐙 **Çoklu Model**: tablo soru cevaplama, optik karakter tanıma, taranmış belgelerden bilgi çıkarma, video sınıflandırma ve görsel soru cevaplama.
+
+🤗 Transformers, PyTorch, TensorFlow ve JAX arasında çerçeve uyumluluğu sağlar. Bu, bir modelin yaşam döngüsünün her aşamasında farklı bir çerçeve kullanma esnekliği sunar; bir çerçevede üç satır kodla bir modeli eğitebilir ve başka bir çerçevede tahminleme için kullanabilirsiniz. Modeller ayrıca üretim ortamlarında kullanılmak üzere ONNX ve TorchScript gibi bir formata aktarılabilir.
```
Comment (merveenoyan) [inline docs/source/tr/index.md:23]:
```suggestion
🤗 Transformers, PyTorch, TensorFlow ve JAX arasında çerçeve (framework) uyumluluğu sağlar. Bu, bir modelin yaşam döngüsünün her aşamasında farklı bir çerçeve kullanma esnekliği sunar; bir çerçevede üç satır kodla bir modeli eğitebilir ve başka bir çerçevede tahminleme için kullanabilirsiniz. Modeller ayrıca üretim ortamlarında kullanılmak üzere ONNX ve TorchScript gibi bir formata aktarılabilir.
```

File: docs/source/tr/index.md [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+    <img alt="HuggingFace Uzman Hızlandırma Programı" src="https://cdn-media.huggingface.co/marketing/transformers/new-support-improved.png" style="width: 100%; max-width: 600px; border: 1px solid #eee; border-radius: 4px; box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.05);">
+</a>
+
+## İçindekiler
+
+Dokümantasyon, beş bölüme ayrılmıştır:
+
+- **BAŞLARKEN**, kütüphanenin hızlı bir turunu ve çalışmaya başlamak için kurulum talimatlarını sağlar.
+- **ÖĞRETİCİLER**, başlangıç yapmak için harika bir yerdir. Bu bölüm, kütüphane kullanmaya başlamak için ihtiyacınız olan temel becerileri kazanmanıza yardımcı olacaktır.
+- **NASIL YAPILIR KILAVUZLARI**, önceden eğitilmiş bir modeli dil modellemesi için ince ayar (finetuning) yapmak veya özel bir model yazmak ve paylaşmak gibi belirli bir hedefe nasıl ulaşılacağını gösterir.
```
Comment (merveenoyan) [inline docs/source/tr/index.md:39]:
```suggestion
- **NASIL YAPILIR KILAVUZLARI**, önceden eğitilmiş bir modele dil modellemesi için ince ayar (fine-tuning) yapmak veya özel bir model yazmak, ve paylaşmak gibi belirli bir hedefe nasıl ulaşılacağını gösterir.
```

File: docs/source/tr/index.md [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+- **KAVRAMSAL REHBERLER**, modellerin, görevlerin ve 🤗 Transformers tasarım felsefesinin temel kavramları ve fikirleri hakkında daha fazla tartışma ve açıklama sunar.
+- **API** tüm sınıfları (class) ve fonksiyonları (functions) açıklar:
+
+  - **ANA SINIFLAR**, yapılandırma, model, tokenizer ve pipeline gibi en önemli sınıfları (classes) ayrıntılandırır.
+  - **MODELLER**, kütüphanede kullanılan her modelle ilgili sınıfları ve fonksiyonları ayrıntılandırır.
+  - **DAHİLİ YARDIMCILAR**, kullanılan yardımcı sınıfları ve fonksiyonları ayrıntılandırır.
```
Comment (merveenoyan) [inline docs/source/tr/index.md:45]:
```suggestion
  - **DAHİLİ YARDIMCILAR**, kullanılan yardımcı sınıfları ve fonksiyonları detaylı olarak inceler.
```

File: docs/source/tr/index.md [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+Dokümantasyon, beş bölüme ayrılmıştır:
+
+- **BAŞLARKEN**, kütüphanenin hızlı bir turunu ve çalışmaya başlamak için kurulum talimatlarını sağlar.
+- **ÖĞRETİCİLER**, başlangıç yapmak için harika bir yerdir. Bu bölüm, kütüphane kullanmaya başlamak için ihtiyacınız olan temel becerileri kazanmanıza yardımcı olacaktır.
+- **NASIL YAPILIR KILAVUZLARI**, önceden eğitilmiş bir modeli dil modellemesi için ince ayar (finetuning) yapmak veya özel bir model yazmak ve paylaşmak gibi belirli bir hedefe nasıl ulaşılacağını gösterir.
+- **KAVRAMSAL REHBERLER**, modellerin, görevlerin ve 🤗 Transformers tasarım felsefesinin temel kavramları ve fikirleri hakkında daha fazla tartışma ve açıklama sunar.
+- **API** tüm sınıfları (class) ve fonksiyonları (functions) açıklar:
+
+  - **ANA SINIFLAR**, yapılandırma, model, tokenizer ve pipeline gibi en önemli sınıfları (classes) ayrıntılandırır.
+  - **MODELLER**, kütüphanede kullanılan her modelle ilgili sınıfları ve fonksiyonları ayrıntılandırır.
```
Comment (merveenoyan) [inline docs/source/tr/index.md:44]:
```suggestion
  - **MODELLER**, kütüphanede kullanılan her modelle ilgili sınıfları ve fonksiyonları detaylı olarak inceler.
```

File: docs/source/tr/index.md [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+  - **DAHİLİ YARDIMCILAR**, kullanılan yardımcı sınıfları ve fonksiyonları ayrıntılandırır.
+
+## Desteklenen Modeller ve Çerçeveler (Frameworks)
```
Comment (merveenoyan) [inline docs/source/tr/index.md:47]:
```suggestion
## Desteklenen Modeller ve Çerçeveler
```

File: docs/source/tr/index.md [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+
+Aşağıdaki tablo, her bir model için kütüphanedeki mevcut desteği temsil etmektedir. Her bir model için bir Python tokenizer'ına ("slow" olarak adlandırılır) sahip olup olmadıkları, 🤗 Tokenizers kütüphanesi tarafından desteklenen hızlı bir tokenizer'a sahip olup olmadıkları, Jax (Flax aracılığıyla), PyTorch ve/veya TensorFlow'da destek olup olmadıklarını göstermektedir.
```
Comment (merveenoyan) [inline docs/source/tr/index.md:49]:
```suggestion
Aşağıdaki tablo, her bir model için kütüphanede yer alan mevcut desteği temsil etmektedir. Her bir model için bir Python tokenizer'ına ("slow" olarak adlandırılır) sahip olup olmadıkları, 🤗 Tokenizers kütüphanesi tarafından desteklenen hızlı bir tokenizer'a sahip olup olmadıkları, Jax (Flax aracılığıyla), PyTorch ve/veya TensorFlow'da destek olup olmadıklarını göstermektedir.
```

File: docs/source/tr/index.md [hunk 10]
```diff
@@ -0,0 +1,295 @@
...
+Lisansa tabi olmayan durumlarda veya yazılı anlaşma olmadıkça, Lisans kapsamında dağıtılan yazılım, herhangi bir türde (açık veya zımni) garanti veya koşul olmaksızın, "OLDUĞU GİBİ" ESASINA GÖRE dağıtılır. Lisans hükümleri, özel belirli dil kullanımı, yetkileri ve kısıtlamaları belirler.
+
+⚠️ Bu dosya Markdown biçimindedir, ancak belge oluşturucumuz için özgü sözdizimleri içerir (MDX gibi) ve muhtemelen Markdown görüntüleyicinizde düzgün bir şekilde görüntülenmeyebilir.
+-->
+
+# 🤗 Transformers
+
+[PyTorch](https://pytorch.org/), [TensorFlow](https://www.tensorflow.org/) ve [JAX](https://jax.readthedocs.io/en/latest/) için State-of-the-art Makine Öğrenimi.
+
+🤗 Transformers, güncel önceden eğitilmiş (pretrained) modelleri indirmenizi ve eğitmenizi kolaylaştıran API'ler ve araçlar sunar. Önceden eğitilmiş modeller kullanarak, hesaplama maliyetlerinizi, karbon ayak izinizi azaltabilir ve sıfırdan bir modeli eğitmek için gereken zaman ve kaynaklardan tasarruf edebilirsiniz. Bu modeller farklı modalitelerde ortak görevleri destekler, örneğin:
```
Comment (merveenoyan) [inline docs/source/tr/index.md:16]:
```suggestion
🤗 Transformers, güncel önceden eğitilmiş (pretrained) modelleri indirmenizi ve eğitmenizi kolaylaştıran API'ler ve araçlar sunar. Önceden eğitilmiş modeller kullanarak, hesaplama maliyetlerinizi ve karbon ayak izinizi azaltabilir, ve sıfırdan bir modeli eğitmek için gereken zaman ve kaynaklardan tasarruf edebilirsiniz. Bu modeller farklı modalitelerde ortak görevleri destekler. Örneğin:
```

**Review by merveenoyan — APPROVED** [review]
Teşekkürler!

**Review by merveenoyan — APPROVED** [review]
LGTM @MKhalusova

**Comment by MKhalusova** [comment]
Thank you for your work, we can merge it now!

**Pull request merged**